src/amber/router/request.cr

```diff
@@ -0,0 +1,55 @@
+require "./params"
+require "./parser"
+require "./file"
+
+class HTTP::Request
```

Comment:
This is out of the scope of this pull, I think, but monkeypatching `HTTP::Request` makes me uncomfortable. I'd much rather create a wrapper object around it and delegate into `HTTP::Request` as needed. Doing this is hostile to having more than one http framework in the same project.

Reply:
Please provide some examples. The reason why I went this route was to avoid initializing an extra class on every request it seems more of an overhead to accomplish the same, the only method that get overriden is the `method` everything else is Amber specific. I wish there was a way to just inherit from HTTP::Request and have HTTP::Context take the new request object but that's like rewriting your HTTP library again..

Reply:
I agree with @eliasjpr here. HTTP::Request is initialized by HTTP::Server, so if we were to create a wrapper object we'd have to monkey patch http server as well.